packages/open-next/src/core/routing/i18n/index.ts

```diff
@@ -39,9 +65,16 @@ export function detectLocale(
     defaultLocale: i18n.defaultLocale,
   });
 
-  return cookiesLocale ?? preferredLocale ?? i18n.defaultLocale;
+  const domainLocale = detectDomainLocale({
+    hostname: internalEvent.headers.host,
```

Comment:
Will that contain the expected value (ref https://github.com/opennextjs/opennextjs-aws/issues/753)

Reply:
^ maybe add that as a comment?